subsys/testsuite/ztest/include/zephyr/ztest_assert.h

```diff
@@ -476,6 +549,149 @@ static inline bool z_zassume(bool cond, const char *default_msg, const char *fil
+ *
+ * @param cond Condition to check
+ * @param ... Optional message and variables to print if the expectation fails
+ */
+#define zexpect_true(cond, ...) zexpect(cond, #cond " is false", ##__VA_ARGS__)
+
+/**
+ * @brief Expect that @a cond is false, otherwise mark test as failed but continue its execution.
+ *
+ * @param cond Condition to check
+ * @param ... Optional message and variables to print if the expectation fails
+ */
+#define zexpect_false(cond, ...) zexpect(!(cond), #cond " is true", ##__VA_ARGS__)
+
+/**
+ * @brief Expect that @a cond is 0 (success), otherwise mark test as failed but continue its
+ * execution.
+ *
+ * @param cond Condition to check
+ * @param ... Optional message and variables to print if the expectation fails
+ */
+#define zexpect_ok(cond, ...) zexpect(!(cond), #cond " is non-zero", ##__VA_ARGS__)
+
+/**
+ * @brief Expect that @a ptr is NULL, otherwise mark test as failed but continue its execution.
+ *
+ * @param ptr Pointer to compare
+ * @param ... Optional message and variables to print if the expectation fails
+ */
+#define zexpect_is_null(ptr, ...) zexpect((ptr) == NULL, #ptr " is not NULL", ##__VA_ARGS__)
+
+/**
+ * @brief Expect that @a ptr is not NULL, otherwise mark test as failed but continue its execution.
+ *
+ * @param ptr Pointer to compare
+ * @param ... Optional message and variables to print if the expectation fails
+ */
+#define zexpect_not_null(ptr, ...) zexpect((ptr) != NULL, #ptr " is NULL", ##__VA_ARGS__)
+
+/**
+ * @brief Expect that @a a equals @a b, otherwise mark test as failed but continue its execution.
+ * expectation fails, the test will be marked as "skipped".
+ *
+ * @param a Value to compare
+ * @param b Value to compare
+ * @param ... Optional message and variables to print if the expectation fails
+ */
+#define zexpect_equal(a, b, ...) zexpect((a) == (b), #a " not equal to " #b, ##__VA_ARGS__)
+
+/**
+ * @brief Expect that @a a does not equal @a b, otherwise mark test as failed but continue its
+ * execution.
+ *
+ * @a a and @a b won't be converted and will be compared directly.
+ *
+ * @param a Value to compare
+ * @param b Value to compare
+ * @param ... Optional message and variables to print if the expectation fails
+ */
+#define zexpect_not_equal(a, b, ...) zexpect((a) != (b), #a " equal to " #b, ##__VA_ARGS__)
+
+/**
+ * @brief Expect that @a a equals @a b, otherwise mark test as failed but continue its execution.
+ *
+ * @a a and @a b will be converted to `void *` before comparing.
+ *
+ * @param a Value to compare
+ * @param b Value to compare
+ * @param ... Optional message and variables to print if the expectation fails
+ */
+#define zexpect_equal_ptr(a, b, ...)                                                               \
+	zexpect((void *)(a) == (void *)(b), #a " not equal to " #b, ##__VA_ARGS__)
+
+/**
+ * @brief Expect that @a a is within @a b with delta @a d, otherwise mark test as failed but
+ * continue its execution.
+ *
+ * @param a Value to compare
```

Comment:
As noted in the test file, if it's idiomatic here, I think you should make these names more obvious.

Reply:
You mean besides a, b, and d? Like rename d to be "delta" or "difference"?

Reply:
Yeah

Reply:
Done. Also did the same to `zexpect_between_inclusive`